src/components/data-management/CellLevelUploadModal.jsx

```diff
@@ -0,0 +1,162 @@
+    newArray.splice(fileIdx, 1);
+    setFilesList(newArray);
+  };
+
+  return (
+    <Modal
+      title=''
+      visible
+      onCancel={onCancel}
+      width='50%'
+      footer={(
+        <Button
+          type='primary'
+          key='create'
+          block
+          disabled={!canUpload}
+          onClick={() => {
+            onUpload(filesList[0]);
+            setFilesList([]);
+          }}
+        >
+          Upload
+        </Button>
+      )}
+    >
+      <Row style={{ margin: '1rem 0' }}>
+        <Col span={24}>
+          <Title level={4}>
+            File upload Upload:
+            <span style={{ color: 'red', marginRight: '2em' }}>*</span>
+          </Title>
+          <Paragraph>
+            Upload a single file containing the metadata in key-value tab-separated format (.tsv)
+          </Paragraph>
+        </Col>
+      </Row>
+      <Row>
+        <Col span={24}>
+          <Dropzone data-testid='dropzone' onDrop={onDrop}>
+            {({ getRootProps, getInputProps }) => (
+              <div
+                style={{ border: '1px solid #ccc', padding: '2rem 0' }}
+                {...getRootProps({ className: 'dropzone' })}
+                id='dropzone'
+              >
+                <input data-testid='drop-input' {...getInputProps()} />
+                <Empty description='Drag and drop the metadata file here or click to browse.' image={Empty.PRESENTED_IMAGE_SIMPLE} />
+              </div>
+            )}
+          </Dropzone>
+        </Col>
+      </Row>
+      <Row>
+        <Col span={24}>
+          {/* eslint-enable react/jsx-props-no-spreading */}
+
+          {filesList.length ? (
+            <>
+              <Divider orientation='center'>To upload</Divider>
+              <List
+                dataSource={filesList}
+                size='small'
+                itemLayout='horizontal'
+                grid='{column: 4}'
+                renderItem={(file) => (
+
+                  <List.Item
+                    key={file.name}
+                    style={{ width: '100%' }}
+                  >
+                    <Space>
+
+                      <CheckCircleTwoTone twoToneColor='#52c41a' />
+
+                      <Text
+                        ellipsis={{ tooltip: file.name }}
+                        style={{ width: '200px' }}
+                      >
+                        {file.name}
+
+                      </Text>
+                      <DeleteOutlined style={{ color: 'crimson' }} onClick={() => { removeFile(file.name); }} />
+                    </Space>
+                  </List.Item>
+
+                )}
+              />
+            </>
+          ) : ''}
+        </Col>
+      </Row>
+    </Modal>
+
+  );
+};
+
+CellLevelUploadModal.propTypes = {
+  onUpload: PropTypes.func,
+  onCancel: PropTypes.func,
+  currentSelectedTech: PropTypes.string,
+};
+
+CellLevelUploadModal.defaultProps = {
+  onUpload: null,
```

Comment:
default prop is null, but wouldn't it crash if it isn't passed when you try to call the function?

Either set a default empty function or perhaps leave it as `isRequired`, since we have onyk one use case that does pass a function we don't need it to have a default value

Reply:
Also check on the other 2 defualt props